elastic/kibana
# [Fleet] Support logstash as an output type in API and Kibana config

## Summary

Resolve #125792 

Add support for logasth type in Fleet ouput API and preconfiguration.

The logstash output require the host to not contains the schema, and allow to configure SSL options.

That PR add a new `ssl` key to the output saved object that is not indexed


## Test 

The API changes are covered by API integration tests


## How to test

You can create a logstash output by adding the following in your Kibana config

```
xpack.fleet.outputs:
  - id: test123
    name: Test logstash
    hosts: [test.fr:443]
    type: "logstash"
    is_default: true
    is_default_monitoring: true
    ssl: 
      certificate_authorities: ['REPLACE_CACERT1']
      certificate: 'REPLACE_CERT'
      key: 'REPLACE_KEY'
```

You should see the output correctly created in Kibana and used in the agent policy (it's not editable for now the UI will come in a following PR)
<img width="1366" alt="Screen Shot 2022-02-21 at 8 56 37 AM" src="https://user-images.githubusercontent.com/1336873/154971049-7aa25a40-e51d-4004-904b-d0d640ce4000.png">



@lykkin This should allow us to start E2E testing with the Agent.
